app/app.py

```diff
@@ -367,6 +381,79 @@ def show_disclaimer() -> None:
     st.button("I Accept", on_click=accept_disclaimer)
 
 
+def show_data_collection_permission() -> None:
+    """Show data sharing consent document on the sidebar."""
+    st.title("Data Sharing Consent")
+    with open("app/data_sharing_consent.txt") as f:
+        data_sharing_consent_text = f.read()
+    st.sidebar.markdown(data_sharing_consent_text)
+
+    st.button("Sure, I'm happy to share!", on_click=accept_data_sharing_consent)
```

Comment:
Nit: there could be a single function for `on_click`, similar to your `post_question_response_to_metric_service`, e.g., `st.button('Happy to share', on_click=data_sharing_consent, args=(True))`